tests/integration/cmd_dev_test.go

```diff
@@ -386,6 +386,47 @@ var _ = Describe("odo dev command tests", func() {
 				})
 			})
 		})
+
+		When("a delay is necessary for the component to start and running odo dev", func() {
+
+			var devSession helper.DevSession
+			var ports map[string]string
+
+			BeforeEach(func() {
+				helper.ReplaceString(filepath.Join(commonVar.Context, "devfile.yaml"), "npm start", "sleep 20 ; npm start")
+
+				var err error
+				devSession, _, _, ports, err = helper.StartDevMode()
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			AfterEach(func() {
+				devSession.Kill()
+				devSession.WaitEnd()
+			})
+
+			It("should first fail then succeed querying endpoint", func() {
+				url := fmt.Sprintf("http://%s", ports["3000"])
+				_, err := http.Get(url)
+				Expect(err).To(HaveOccurred())
+
+				podName := commonVar.CliRunner.GetRunningPodNameByComponent(cmpName, commonVar.Project)
+				Eventually(func() bool {
+					logs := helper.GetCliRunner().GetLogs(podName)
```

Comment:
Can we use `odo logs` here instead of `kubectl`/`oc`?

Reply:
No, we discussed to not rely on other odo commands during integration tests. This way, odo logs can be buggy, or be removed, and this won't affect this test